test/s3select_test.cpp

```diff
@@ -3392,3 +3392,288 @@ TEST(TestS3selectFunctions, test_trim_expressions)
 
   ASSERT_EQ(s3select_result_3, s3select_result_4);
 }
+
+TEST(TestS3selectFunctions, truefalse)
+{
+  test_single_column_single_row("select 2 from s3object where true or false;","2,\n");
+  test_single_column_single_row("select 2 from s3object where true or true;","2,\n");
+  test_single_column_single_row("select 2 from s3object where null or true ;","2,\n");
+  test_single_column_single_row("select 2 from s3object where true and true;","2,\n");
+  test_single_column_single_row("select 2 from s3object where true == true ;","2,\n");
+  test_single_column_single_row("select 2 from stdin where 1<2 == true;","2,\n");
+  test_single_column_single_row("select 2 from stdin where 1==1 == true;","2,\n");
+  test_single_column_single_row("select 2 from stdin where false==false == true;","2,\n");
+  test_single_column_single_row("select 2 from s3object where false or true;","2,\n");
+  test_single_column_single_row("select true,false from s3object where false == false;","true,false,\n");
+  test_single_column_single_row("select count(*) from s3object where not (1>2) == true;","1,");
+  test_single_column_single_row("select count(*) from s3object where not (1>2) == (not false);","1,");
+  test_single_column_single_row("select (true or false) from s3object;","true,\n");
+  test_single_column_single_row("select (true and true) from s3object;","true,\n");
+  test_single_column_single_row("select (true and null) from s3object;","null,\n");
+  test_single_column_single_row("select (false or false) from s3object;","false,\n");
+  test_single_column_single_row("select (not true) from s3object;","false,\n");
+  test_single_column_single_row("select (not 1 > 2) from s3object;","true,\n");
+  test_single_column_single_row("select (1 > 2) from s3object;","false,\n");
+}
+
+TEST(TestS3selectFunctions, boolcast)
+{
+  test_single_column_single_row("select cast(5 as bool) from s3object;","true,\n");
+  test_single_column_single_row("select cast(0 as bool) from s3object;","false,\n");
+  test_single_column_single_row("select cast(true as bool) from s3object;","true,\n");
+  test_single_column_single_row("select cast('a' as bool) from s3object;","false,\n");
+}
+
+TEST(TestS3selectFunctions, predicate_as_projection_column)
```

Comment:
ok,there is a good coverage for predicate as projection.

the when-case statements are projection which contain predicates, thus

it's important to have complex query, contains /when-case/ statement and another predicate as projection, it will test correctness of the AST and also the syntax-definitions.